aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/RequestRetry.kt

```diff
@@ -0,0 +1,45 @@
+package com.amplifyframework.datastore.syncengine
+
+import com.amplifyframework.AmplifyException
+
+import io.reactivex.rxjava3.core.Single
+import io.reactivex.rxjava3.core.SingleEmitter
+import java.util.concurrent.TimeUnit
+import kotlin.math.pow
+
+
+class RequestRetry(private val maxExponent: Int = 8,
+                    private val jitterFactor: Int = 100,
+                    private val maxAttempts: Int = 3) {
+
+    private var numberOfAttempts = 0
+
+
+        fun <T> retry(single: Single<T>, skipExceptions: List<Class<out Throwable>?>): Single<T> {
+            return Single.create { emitter -> call(single, emitter, 0, skipExceptions) }
+        }
+
+        private fun <T> call(single: Single<T>, emitter: SingleEmitter<T>, delayInSeconds: Long, skipExceptions: List<Class<out Throwable>?> ) {
+           single.delaySubscription(delayInSeconds, TimeUnit.SECONDS)
+                .subscribe({
+                            t: T -> emitter.onSuccess(t)
```

Comment:
I believe this can be simplified to just use the implied `it` variable.  This is the recommendation from the [Kotlin coding conventions](https://kotlinlang.org/docs/coding-conventions.html#lambda-parameters).

```suggestion
                            emitter.onSuccess(it)
```

Same thing with the error handler below (use implied `it`)

Reply:
Actually, I think you can simplify it even further, and make this just `emitter::onSuccess`